katharsis-core/src/main/java/io/katharsis/resource/registry/repository/adapter/ResponseRepositoryAdapter.java

```diff
@@ -0,0 +1,348 @@
+    	}
+   		return enrichedLinksInformation;
+	}
+
+    private LinksInformation enrichPageLinksInformation(LinksInformation linksInformation, Iterable<?> resources,
+			QueryAdapter queryAdapter, RepositoryRequestSpec requestSpec) {
+    	if(!(resources instanceof PagedResultList) || queryAdapter.getLimit() == null){
+    		return linksInformation;
+    	}
+    
+		if(linksInformation != null && !(linksInformation instanceof PagedLinksInformation)){
+			throw new IllegalStateException(linksInformation + " must implement " + PagedLinksInformation.class.getName() + " to support pagination link computation with " + PagedResultList.class);
+		}
+		
+		// only enrich if not already set
+		PagedLinksInformation pagedLinksInformation = (PagedLinksInformation) linksInformation;
+		if(pagedLinksInformation == null){
+			// use default implementation if no link information provided by repository
+			pagedLinksInformation = new DefaultPagedLinksInformation();
+		}
+		if(!hasPageLinks(pagedLinksInformation)){
+			PagedResultList<?> pageResultList = (PagedResultList<?>) resources;
+			doEnrichPageLinksInformation(pagedLinksInformation, pageResultList, queryAdapter, requestSpec);
+		}
+		return pagedLinksInformation;
+	}
+
+	private boolean hasPageLinks(PagedLinksInformation pagedLinksInformation) {
+		return pagedLinksInformation.getFirst() != null || pagedLinksInformation.getLast() != null || pagedLinksInformation.getPrev() != null || pagedLinksInformation.getNext() != null;
+	}
+
+	private void doEnrichPageLinksInformation(PagedLinksInformation pagedLinksInformation, PagedResultList<?> pageResultList,
+			QueryAdapter queryAdapter, RepositoryRequestSpec requestSpec) {
+    	long total = pageResultList.getTotalCount();
+		long pageSize = queryAdapter.getLimit().longValue();
+		long offset = queryAdapter.getOffset();
+
+		long currentPage = offset / pageSize;
+		if (currentPage * pageSize != offset) {
+			throw new IllegalArgumentException("offset " + offset + " is not a multiple of limit " + pageSize);
+		}
+		long totalPages = (total + pageSize - 1) / pageSize;
+
+		QueryAdapter pageSpec = queryAdapter.duplicate();
+		pageSpec.setLimit(pageSize);
+
+		if(totalPages > 0){
+			pageSpec.setOffset(0);
+			pagedLinksInformation.setFirst(toUrl(pageSpec, pageResultList, requestSpec));
+	
+			pageSpec.setOffset((totalPages - 1) * pageSize);
+			pagedLinksInformation.setLast(toUrl(pageSpec, pageResultList, requestSpec));
+	
+			if (currentPage > 0) {
+				pageSpec.setOffset((currentPage - 1) * pageSize);
+				pagedLinksInformation.setPrev(toUrl(pageSpec, pageResultList, requestSpec));
+			}
+	
+			if (currentPage < totalPages - 1) {
+				pageSpec.setOffset((currentPage + 1) * pageSize);
+				pagedLinksInformation.setNext(toUrl(pageSpec, pageResultList, requestSpec));
+			}		
+		}
+	}
+	
+	private <T> String toUrl(QueryAdapter queryAdapter, PagedResultList<T> pagedResultList, RepositoryRequestSpec requestSpec) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "pagedResultList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)